File: Src/Icm.Core/Configuration/TypedValue.vb

Namespace Icm.Configuration

    Public Class TypedValue
        Public ReadOnly Type As Type
        Public ReadOnly Value As Object

        Public Sub New(ByVal t As Type, ByVal val As Object)
            Debug.Assert(val Is Nothing OrElse val.GetType Is t)
            Type = t
            Value = val
        End Sub
    End Class

End Namespace

File: Src/Icm.Core/Configuration/TypedValue.cs
using System;
using System.Diagnostics;

namespace Icm.Configuration
{

	public class TypedValue
	{
		public readonly Type Type;

		public readonly object Value;
		public TypedValue(Type t, object val)
		{
			Debug.Assert(val == null || ReferenceEquals(val.GetType(), t));
			Type = t;
			Value = val;
		}
	}

}